index.bs

```diff
@@ -75,6 +75,16 @@ interface LanguageModelParams {
   readonly attribute float maxTemperature;
 };
 
+// A description of a funnction call that a language model can invoke.
+dictionary LanguageModelTool {
+  required DOMString name;
+  required DOMString description;
+  // JSON schema for the input parameters.
+  required object inputSchema;
+  // The function to be invoked by user agent on behalf of language model.
+  required Function execute;
```

Comment:
Let's define this to return a promise for a string, as it will make the bindings slightly nicer.

Define a callback type like

```webidl
callback LanguageModelToolFunction = Promise<DOMString> (any... arguments);
```

and then use `LanguageModelToolFunction` instead of `Function`.